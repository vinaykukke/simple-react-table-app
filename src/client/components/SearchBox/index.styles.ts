import styled from 'styled-components';

export const Input = styled.input`
  height: 30px;
  width: 100%;
  border-radius: 5px;
  border: none;
  padding: 10px;
  background: #e4e4e4;
`;

export const SearchboxContainer = styled.div`
  max-width: 60%;
  margin: 1em 0;

  > div {
    color: white;
    margin: 1em 0;
    font-size: 12px;
    font-style: italic;
  }
`;
