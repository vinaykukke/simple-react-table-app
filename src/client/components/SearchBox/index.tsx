import React from 'react';
import Button from 'Components/Button';
import * as S from './index.styles';

const SearchBox = (props: any) => {
  return <S.SearchboxContainer>
    <S.Input
      type='text'
      placeholder='Search here'
      {...props}
    />
    <div>This search is case sensitive</div>
    <div>
      NOTE: The pagination is actually working. But since there are only 5 unique items and the other 25 are repeated
      it seems like the pagination does not work. You can test this out by changing the `defaultPageSize` prop in
      `src/client/components/SearchBox/index.tsx` to an odd number like 7. OR. even just select show `20 rows` and click the next button.
    </div>
  </S.SearchboxContainer>;
};

export default SearchBox;
