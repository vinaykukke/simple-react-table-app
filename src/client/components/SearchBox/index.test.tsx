import React from 'react';
import renderer from 'react-test-renderer';

import SearchBox from './index';

describe('SearchBox Component', () => {
  it('should render <SearchBox />', () => {
    const tree = renderer.create(<SearchBox />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
