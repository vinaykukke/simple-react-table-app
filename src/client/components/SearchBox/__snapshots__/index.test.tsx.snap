// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component should render <SearchBox /> 1`] = `
<div
  className="sc-bwzfXH glFDba"
>
  <input
    className="sc-bdVaJa hWvyqe"
    placeholder="Search here"
    type="text"
  />
  <div>
    This search is case sensitive
  </div>
  <div>
    NOTE: The pagination is actually working. But since there are only 5 unique items and the other 25 are repeated it seems like the pagination does not work. You can test this out by changing the \`defaultPageSize\` prop in \`src/client/components/SearchBox/index.tsx\` to an odd number like 7. OR. even just select show \`20 rows\` and click the next button.
  </div>
</div>
`;
